functional_tests.go

```diff
@@ -121,6 +121,31 @@ func ignoredLog(function string, args map[string]interface{}, startTime time.Tim
 	return log.WithFields(fields)
 }
 
+// Delete objects and buckets recursively
+func cleanupBuckets(c *minio.Client) (err error) {
+
+	doneCh := make(chan struct{})
+	defer close(doneCh)
+
+	buckets, err := c.ListBuckets()
+
+	for _, bucket := range buckets {
+		for obj := range c.ListObjects(bucket.Name, "", true, doneCh) {
```

Comment:
you can put each ListObjects() on a go-routine since they are independent buckets and clean them up.